test/test.js

```diff
@@ -0,0 +1,261 @@
+'use strict';
+
+var test = require('tape');
+var MapboxGLScale = require('../');
+
+test('constant', function(t) {
+    t.test('array', function(t) {
+        var scale = MapboxGLScale([1]);
+
+        t.deepEqual(scale(0), [1]);
+        t.deepEqual(scale(1), [1]);
+        t.deepEqual(scale(2), [1]);
+
+        t.end();
+    });
+
+    t.test('number', function(t) {
+        var scale = MapboxGLScale(1);
+
+        t.equal(scale(0), 1);
+        t.equal(scale(1), 1);
+        t.equal(scale(2), 1);
+
+        t.end();
+    });
+
+    t.end();
+});
+
+test('domain & range', function(t) {
+    t.test('one element', function(t) {
+        var scale = MapboxGLScale({
+            domain: [1],
+            range: [2]
+        });
+
+        t.equal(scale(0), 2);
+        t.equal(scale(1), 2);
+        t.equal(scale(2), 2);
+
+        t.end();
+    });
+
+    t.test('two elements', function(t) {
+        var scale = MapboxGLScale({
+            domain: [1, 3],
+            range: [2, 6]
+        });
+
+        t.equal(scale(0), 2);
+        t.equal(scale(1), 2);
+        t.equal(scale(2), 4);
+        t.equal(scale(3), 6);
+        t.equal(scale(4), 6);
+
+        t.end();
+    });
+
+    t.test('three elements', function(t) {
+        var scale = MapboxGLScale({
+            domain: [1, 3, 5],
+            range: [2, 6, 10]
+        });
+
+        t.equal(scale(0), 2);
+        t.equal(scale(1), 2);
+        t.equal(scale(2), 4);
+        t.equal(scale(3), 6);
+        t.equal(scale(4), 8);
+        t.equal(scale(5), 10);
+        t.equal(scale(6), 10);
+
+        t.end();
+    });
+
+});
+
+test('base', function(t) {
+    var scale = MapboxGLScale({
+        domain: [1, 3],
+        range: [2, 6],
+        base: 2
+    });
+
+    t.equal(scale(0), 2);
+    t.equal(scale(1), 2);
+    t.equal(scale(2), 30 / 9);
+    t.equal(scale(3), 6);
+    t.equal(scale(4), 6);
+
+    t.end();
+});
+
+test('property', function(t) {
```

Comment:
Add tests for:

* string-valued domains
* boolean-valued domains
* `Scale({domain: [1], range: [2], property: 'p'})({p: 'omg'})`
* `Scale({domain: ['A', 'B', 'C'], range: [1, 2, 3], property: 'p'})({p: 42})`
* `Scale({domain: ['A', 'B', 'C'], range: [1, 2, 3], property: 'p'})({p: 'D'})`

What other corner cases are there?